share/janitor/tasks.py

```diff
@@ -2,19 +2,40 @@
 
 import celery
 
+from django.db.models import Exists
+from django.db.models import OuterRef
+
 from share import tasks
 from share.models import RawDatum
+from share.models import NormalizedData
 
 
 logger = logging.getLogger(__name__)
 
 
 @celery.shared_task(bind=True)
-def rawdata_janitor(self, limit=100):
+def rawdata_janitor(self, limit=500):
     """Find RawDatum that do not have a NormalizedData process them
     """
     count = 0
-    for rd in RawDatum.objects.select_related('suid__source_config').filter(normalizeddata__isnull=True).order_by('id')[:limit].iterator():
+
+    # NOTE: Do NOT use .iterator here. It will create a temporary table and eat disk space like no other
+    # the limit lets this query fit nicely in memory and actually finish executing
+    # Be very careful about changing this query. If you do change it, make sure the EXPLAIN looks something like this:
```

Comment:
For the old query (with `filter(normalizeddata_isnull=True)`), I get an EXPLAIN pretty close to what you put here:
```
Limit  (cost=0.88..13803.64 rows=363 width=1611)
  ->  Nested Loop  (cost=0.88..13803.64 rows=363 width=1611)
        ->  Nested Loop  (cost=0.74..13740.73 rows=363 width=1195)
              ->  Merge Left Join  (cost=0.44..13609.28 rows=363 width=1163)
                    Merge Cond: (share_rawdatum.id = share_normalizeddata.raw_id)
                    Filter: (share_normalizeddata.id IS NULL)
                    ->  Index Scan using share_rawdata_pkey on share_rawdatum  (cost=0.29..13303.84 rows=72640 width=1163)
                    ->  Index Scan using share_normalizeddata_raw_id_b0718619 on share_normalizeddata  (cost=0.15..102.16 rows=1734 width=8)
              ->  Index Scan using share_sourceuniqueidentifier_pkey on share_sourceuniqueidentifier  (cost=0.29..0.35 rows=1 width=32)
                    Index Cond: (id = share_rawdatum.suid_id)
        ->  Index Scan using share_sourceconfig_pkey on share_sourceconfig  (cost=0.14..0.16 rows=1 width=416)
              Index Cond: (id = share_sourceuniqueidentifier.source_config_id)
```
but for the new query (with `annotate`), I get something different:
```
Limit  (cost=18.35..6510.74 rows=500 width=1611)
  ->  Hash Join  (cost=18.35..471625.27 rows=36320 width=1611)
        Hash Cond: (share_sourceuniqueidentifier.source_config_id = share_sourceconfig.id)
        ->  Merge Join  (cost=0.69..319653.81 rows=36320 width=1195)
              Merge Cond: (share_rawdatum.suid_id = share_sourceuniqueidentifier.id)
              ->  Index Scan using share_rawdatum_suid_id_d6dd4177 on share_rawdatum  (cost=0.29..316583.48 rows=36320 width=1163)
                    Filter: (((NOT no_change) OR (no_change IS NULL)) AND (NOT (alternatives: SubPlan 3 or hashed SubPlan 4)))
                    SubPlan 3
                      ->  Index Only Scan using share_normalizeddata_raw_id_b0718619 on share_normalizeddata u0_2  (cost=0.15..36.31 rows=9 width=0)
                            Index Cond: (raw_id = share_rawdatum.id)
                    SubPlan 4
                      ->  Seq Scan on share_normalizeddata u0_3  (cost=0.00..34.34 rows=1734 width=4)
              ->  Index Scan using share_sourceuniqueidentifier_pkey on share_sourceuniqueidentifier  (cost=0.29..2460.89 rows=72640 width=32)
        ->  Hash  (cost=15.07..15.07 rows=207 width=416)
              ->  Seq Scan on share_sourceconfig  (cost=0.00..15.07 rows=207 width=416)
        SubPlan 1
          ->  Index Only Scan using share_normalizeddata_raw_id_b0718619 on share_normalizeddata u0  (cost=0.15..36.31 rows=9 width=0)
                Index Cond: (raw_id = share_rawdatum.id)
        SubPlan 2
          ->  Seq Scan on share_normalizeddata u0_1  (cost=0.00..34.34 rows=1734 width=4)
```
Is there something odd about my data, or does the EXPLAIN in this comment not line up with the query?